Snowflake.Data/Core/SFStatement.cs

```diff
@@ -309,19 +366,26 @@ private void registerQueryCancellationCallback(int timeout, CancellationToken ex
 
                 var lastResultUrl = response.data?.getResultUrl;
 
-                while (RequestInProgress(response) || SessionExpired(response))
+                if (asyncExec)
                 {
-                    var req = BuildResultRequest(lastResultUrl);
-                    response = await _restRequester.GetAsync<QueryExecResponse>(req, cancellationToken).ConfigureAwait(false);
-
-                    if (SessionExpired(response))
-                    {
-                        logger.Info("Ping pong request failed with session expired, trying to renew the session.");
-                        await SfSession.renewSessionAsync(cancellationToken).ConfigureAwait(false);
-                    }
-                    else
+                    SfSession.AsyncQueries.Add(response.data.queryId, QueryStatus.RUNNING);

```

Comment:
is it possible that even though we send asyncExec = "false" we are getting the result immediately?

Reply:
I don't believe so, if it's set to false then it uses the original code and waits for the query to finish executing before returning